d2bs/kolbot/libs/bots/BattleOrders.js

```diff
@@ -74,10 +124,18 @@ MainLoop:
 				break MainLoop;
 			}
 
+			if (this.AmTardy()) {
+				break MainLoop;
+			}
+
 			if (getTickCount() - tick >= failTimer * 1000) {
 				showConsole();
 				print("ÿc1BO timeout fail.");
-				quit();
+				if (Config.BattleOrders.QuitOnFailedGet) {
```

Comment:
The other block i meant in

>Why the same block code here, as in the other switch case?
Maybe store a variable saying you failed, and if so quit / break, instead of duplication code